File: _clases/C/02datatypes.c
#include <stdio.h>
#include <stdlib.h>	
#include <math.h>	//aritmetica
#include <string.h>	//strings

#define PI 3.14 	//constantes simbólicas, reemplaza PI por 3.14

int main(){

	printf("PI=%f\n",PI);

	//integers
	printf("INTEGERS:\n");
	printf("---------\n");
	//	op. logicas
	printf("Op. lógicas:\n");
	int p,q;
	
	p=0;
	q=1;

	printf("     p = %d\n",p);
	printf("     q = %d\n",q);
	printf("    !p = %d\n",!p);
	printf(" p & q = %d\n",p&q);
	printf(" p | q = %d\n",p|q);
	printf("p == q = %d\n",p==q);
	printf("p != q = %d\n",p!=q);

	//	op. relacionales
	printf("Op. relacionales:\n");
	int n=-3,m=7;
	
	printf("     n = %d\n",n);
        printf("     m = %d\n",m);
	printf("n == m = %d\n",n==m);
	printf("n != m = %d\n",n!=m);
	printf("n <  m = %d\n",n< m);
	printf("n <= m = %d\n",n<=m);
	printf("n >  m = %d\n",n> m);
	printf("n >= m = %d\n",n>=m);

	//	op. aritmeticas
	printf("Op. artimeticas:\n");
	

	printf("n + m = %d\n",n+m);
	printf("n / m = %d\n",n/m);
	printf("n %% m = %d\n",n%m);
	
	//	op. intrinsecas
	printf("Op. intrinsecas:\n");
	printf("abs(n) = %d\n",abs(n));

	//FLOAT
	printf("FLOATING POINT:\n");
	printf("---------------\n");

	float x, y;
	x=1.2e-1;
	y=-2.78;

	printf("    x = %6.2f\n",x);
	printf("    y = %7.3f\n",y);
	printf("x - y = %7.3f\n",x-y);
	printf("x * y = %7.3f\n",x*y);
	
	double xx, yy;
	xx=1.2e-1;
        yy=-2.78;
	printf("pow(x,y) = %7.3f\n",pow(x,y));
	printf("sqrt(x) = %7.3f\n",sqrt(xx));
	printf("exp(x) = %7.3f\n",exp(xx));
	printf("log(x) = %7.3f\n",log(xx));
	printf("sin(x) = %7.3f\n",sin(xx));
	printf("cos(x) = %7.3f\n",cos(xx));

	//Character
	printf("CHARACTER / STRING:\n");
	printf("-------------------\n");

	char* letra = "a";
	char str1[14]="asdasd asddsa";

	printf("Esto es una letra %s\n",letra);
	printf("Esto es un string %s\n",str1);

/*
	char c;
	c = getchar();
	putchar(c);
        
         character*30       :: str1="Esto es un string       "
         character (len=30) :: str2=", este es otro string.  "
  
         print*, str1
         
         !Operaciones con CHARACTER:
         print*, str1(9:25)
         print*, str1//str2
         print*, trim(str1)//trim(str2)
  
         print*, len(str1)
         print*, len(trim(str1))

         print*, repeat(str2(11:15),4)
         
         print*, scan(str1, 'aeiou')
         print*, index(str2, 'otro',back=.true.)
         print*, verify(str1,'Esto es un strang')
*/
   	char srcString[40];
   	char destString[100];
  
   	memset(destString, '\0', sizeof(destString));
   	strcpy(srcString, "Este es un string");
   	strcpy(destString, srcString);
   	printf("String resultante: %s\n", destString);

	// Type conversion:
	/* bool -> char -> short int -> int -> unsigned int -> long -> unsigned -> long long -> float -> double -> long double 	*/
	// char to int
	char a[1] = "j";
	int i= atoi(a);
	printf("letra: %s; int %d\n", a, atoi(a));



  return 0;
}

File: _clases/FORTRAN/02datatypes.f90
       PROGRAM datatypes
        implicit none
        
        logical  :: p, q=.TRUE.
        
        !Inicialización
        p=.FALSE.
        
        print*,'p: ',p
        print*,'q: ',q
        
        !Operaciones LOGICAS:
        print*,  .NOT. p
        print*,  p .AND.  q
        print*,  p .OR.   q
        print*,  p .EQV.  q
        print*,  p .NEQV. q  !XOR

       END PROGRAM datatypes
	PROGRAM datatypes
	   implicit none
	   integer :: n=7,m
	   
           m=3

	!Operaciones con INTEGER
	!Relacionales (retorna LOGICAL)
           print*, n == m  ! .EQ.
	   print*, n /= m  ! .NE.
	   print*, n <  m  ! .LT.
	   print*, n <= m  ! .LE.
	   print*, n >  m  ! .GT.
	   print*, n >= m  ! .GE.
	
        !Aritméticas: + - * / **
	   print*, n+m
	   print*, n/m

	!Funciones intrinsecas
	   print*, abs(n)   !modulo de n
	   print*, mod(n,m) !resto de n/m

        END PROGRAM datatypes
        PROGRAM datatypes
                implicit none
        
                real 		:: x=1.2e-1, y=2.78
                real, parameter :: pi=3.14159265365897
        
        !Operaciones con REAL
        !Aritméticas: 	+ - * / **
                print*, 3+x*y
                print*, (3+x)*y
                print*, y**2*3
                print*, y**(2*3)
        !Funciones intrinsecas
                print*, sqrt(x),exp(x),log(x),log10(x);
                print*, sin(x),cos(x),tan(x);
                print*, asin(x),acos(x),atan(x);
                
                print*, real(7), int(7.000);
                print*, mod(x,y), abs(x);
                print*, min(x,y), max(x,y);
              
        END PROGRAM datatypes
        PROGRAM datatypes
                implicit none
                complex :: z1=(0,1),z2=(3,2)
        
       !Operaciones con COMPLEX
                print*, z1
                print*, z2
       !Aritmética: 	+ - * / **
                print*, z1+z2
                print*, z1-z2
                print*, z1*z2
                print*, z1/z2
                print*, z1**2
       
        !Funciones intrinsecas
                print*, real(z1)
                print*, aimag(z1)
                print*, conjg(z1)
                print*, abs(z2)
                print*, atan2(real(z2),aimag(z2))
        
        END PROGRAM datatypes
       program datatypes  
      
         implicit none

         character*30       :: str1="Esto es un string       "
         character (len=30) :: str2=", este es otro string.  "
  
         print*, str1
         
         !Operaciones con CHARACTER:
         print*, str1(9:25)
         print*, str1//str2
         print*, trim(str1)//trim(str2)
  
         print*, len(str1)
         print*, len(trim(str1))

         print*, repeat(str2(11:15),4)
         
         print*, scan(str1, 'aeiou')
         print*, index(str2, 'otro',back=.true.)
         print*, verify(str1,'Esto es un strang')
       end program datatypes
